include/realizations/catchment/Bmi_Formulation.hpp

```diff
@@ -201,8 +207,27 @@ namespace realization {
          */
         virtual bool is_model_initialized() = 0;
 
+        /**
+         * Set the precision of output values when converted to text.
+         *
+         * @param precision The desired precision, as a number of decimal places.
+         */
+        void set_output_precision(int precision) {
+            output_precision = precision;
+            output_text_stream = std::make_shared<std::ostringstream>();
```

Comment:
This seems too easy to inadvertently kill a stream before it has been "properly" serialized or whatever, e.g.

```c++
set_output_preceision(5);
output_text_stream << "Hello ";
set_output_preceision(6);
output_text_stream << "World\n";
```
Would leave you with a stream of just "World\n".